File: AspNetMvcUnleashed/0672329980_source/Chapter03/VB/MvcApplication1/MvcApplication1/Selectors/AjaxMethodAttribute.vb
Imports System.Reflection

Public Class AjaxMethodAttribute
    Inherits ActionMethodSelectorAttribute

    Public Overrides Function IsValidForRequest(ByVal controllerContext As ControllerContext, ByVal methodInfo As MethodInfo) As Boolean
        Return controllerContext.HttpContext.Request.IsAjaxRequest
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter03/CS/MvcApplication1/MvcApplication1/Selectors/AjaxMethodAttribute.cs
using System.Reflection;
using System.Web.Mvc;

namespace MvcApplication1.Selectors
{
    public class AjaxMethod : ActionMethodSelectorAttribute
    {
        public override bool IsValidForRequest(ControllerContext controllerContext, MethodInfo methodInfo)
        {
            return controllerContext.HttpContext.Request.IsAjaxRequest();
        }
    }
}
